### PR TITLE
optimize(hz): refuse all new operation if there already exists a project

### LINKED ISSUE #529: hz 命令接口的讨论
**Describe the Question**

主要是一个demo代码里, 直接make了一下, 导致当前的代码, go.mod出现了奇怪的修改, 感觉到对普通用户应该是一个不太友好的体验
所以想提个issue讨论一下

**Reproducible Code**
```
Makefile

init_api:
	hz new -I idl -idl idl/user/user.proto -module github.com/cloudwego/hertz-examples/bizdemo/hertz_gorm_gen
update_api:
	hz update -I idl -idl idl/user/user.proto
```
在已生成的代码里 直接make一下

**Expected behavior**
这里想讨论一下目前hz命令的语义
```
NAME:
   hz - A idl parser and code generator for Hertz projects

USAGE:
   hz [global options] command [command options] [arguments...]

VERSION:
   v0.5.0
COMMANDS:
   new      Generate a new Hertz project
   update   Update an existing Hertz project
   model    Generate model code only
   client   Generate hertz client based on IDL
   help, h  Shows a list of commands or help for one command
```
1. 个人认为new如果是代表创建的意思, 那创建结束了, 不是太应该允许运行重复执行new, 只能update
2. new project 跟 new project code 觉得也分开一下比较合理 
``` 
     hz api all
     hz api model
     hz api middleware
     hz api route
     hz project new
     hz project update
```
我的guan观点是: 像官网demo的Makefile, 都是init_api ==> hz new 这说明其实hz接口定义比较容易造成混淆

**Screenshots**
image

image

image

**Hertz version:**

Please provide the version of Hertz you are using.

**Environment:**

**Additional context**

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?

optimize

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.

- 项目存在时拒绝使用任何 new 操作

#### Which issue(s) this PR fixes:

Fixes https://github.com/cloudwego/hertz/issues/529

### CODE REVIEW

File: cmd/hz/go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/cloudwego/hertz/cmd/hz
 go 1.16
 
 require (
+	github.com/AlecAivazis/survey/v2 v2.3.6
```
Comment (Duslia) [inline cmd/hz/go.mod:6]: If you add e new dependency, please add its license to LICENSE dir

File: cmd/hz/app/app.go [hunk 2]
```diff
@@ -44,6 +46,23 @@ func New(c *cli.Context) error {
 	setLogVerbose(args.Verbose)
 	logs.Debugf("args: %#v\n", args)
 
+	// determine if code needs to be regenerated
```
Comment (Duslia) [inline cmd/hz/app/app.go:49]: I think whether we can regulate user behavior that `hz new` is only used to create a project and if hz detect the project has been created, just exit with error: `the hz project has been created, please use update command to update your project`

File: cmd/hz/app/app.go [hunk 3]
```diff
@@ -44,6 +46,23 @@ func New(c *cli.Context) error {
...
+	exists, err := util.PathExist(path.Join(args.OutDir, meta.ManifestFile))
+	if err != nil {
+		return cli.Exit(err, meta.LoadError)
+	}
+
+	if exists {
+		var isOverride bool
+		qs := overrideQuestions()
+		if err = survey.Ask(qs, &isOverride); err != nil {
```
Comment (welkeyever) [inline cmd/hz/app/app.go:58]: If it is just to achieve this function, there should be no need to introduce this library?
For instance: add a flag to enable override && refuse all new operation if there already exists a project.

Comment (FGYFFFF) [reply cmd/hz/app/app.go]: +1~Interactive methods may block the automatic execution of scripts. Perhaps if there are already existing files in the "new", we can skip them by default and print a warning log, and if the user needs to regenerate them, delete the corresponding files manually

Comment (li-jin-gou) [reply cmd/hz/app/app.go]: done

File: cmd/hz/app/app.go (hunk 2 above)
Comment (li-jin-gou) [reply cmd/hz/app/app.go]: done

File: cmd/hz/app/app.go [hunk 4]
```diff
@@ -44,6 +45,15 @@ func New(c *cli.Context) error {
 	setLogVerbose(args.Verbose)
 	logs.Debugf("args: %#v\n", args)
 
+	exist, err := util.PathExist(path.Join(args.OutDir, meta.ManifestFile))
+	if err != nil {
+		return cli.Exit(err, meta.LoadError)
+	}
+
+	if exist {
+		return cli.Exit(fmt.Errorf("project already exists"), meta.LoadError)
```
Comment (welkeyever) [inline cmd/hz/app/app.go:54]: ”project already exists“ is a little bit simple / maybe add some instruction for users(tell why/how)?

Comment (li-jin-gou) [reply cmd/hz/app/app.go:54]: ok

Comment (FGYFFFF) [reply cmd/hz/app/app.go:54]: add a error info "If you still want to "new" a project please delete the ".hz" manually

Comment (li-jin-gou) [reply cmd/hz/app/app.go:54]: thanks

**Review by Duslia — DISMISSED** [review]

**Review by wzekin — DISMISSED** [review]

File: cmd/hz/app/app.go [hunk 5]
```diff
@@ -44,6 +45,15 @@ func New(c *cli.Context) error {
 	setLogVerbose(args.Verbose)
 	logs.Debugf("args: %#v\n", args)
 
+	exist, err := util.PathExist(path.Join(args.OutDir, meta.ManifestFile))
+	if err != nil {
+		return cli.Exit(err, meta.LoadError)
+	}
+
+	if exist {
+		return cli.Exit(fmt.Errorf("if you still want to new a project please delete the .hz"), meta.LoadError)
```
Comment (welkeyever) [inline cmd/hz/app/app.go:54]: A `--force` flag may better than "delete .hz file by user"?

Comment (li-jin-gou) [reply cmd/hz/app/app.go:54]: GET

**Review by FGYFFFF — APPROVED** [review]

**Comment by li-jin-gou** [comment]
cc @welkeyever @Duslia @wzekin

**Review by wzekin — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**